engine/src/main/java/org/pentaho/di/job/entries/shell/JobEntryShell.java

```diff
@@ -0,0 +1,779 @@
+    try {
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "file_name", filename );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "work_directory", workDirectory );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "arg_from_previous", argFromPrevious );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "exec_per_row", execPerRow );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "set_logfile", setLogfile );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "set_append_logfile", setAppendLogfile );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "add_date", addDate );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "add_time", addTime );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "logfile", logfile );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "logext", logext );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "loglevel", logFileLevel == null ? LogLevel.NOTHING
+        .getCode() : logFileLevel.getCode() );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "insertScript", insertScript );
+      rep.saveJobEntryAttribute( id_job, getObjectId(), "script", script );
+
+      // save the arguments...
+      if ( arguments != null ) {
+        for ( int i = 0; i < arguments.length; i++ ) {
+          rep.saveJobEntryAttribute( id_job, getObjectId(), i, "argument", arguments[i] );
+        }
+      }
+    } catch ( KettleDatabaseException dbe ) {
+      throw new KettleException( "Unable to save job entry of type 'shell' to the repository", dbe );
+    }
+  }
+
+  public void clear() {
+    super.clear();
+
+    filename = null;
+    workDirectory = null;
+    arguments = null;
+    argFromPrevious = false;
+    addDate = false;
+    addTime = false;
+    logfile = null;
+    logext = null;
+    setLogfile = false;
+    execPerRow = false;
+    setAppendLogfile = false;
+    insertScript = false;
+    script = null;
+  }
+
+  /**
+   * @deprecated use {@link #setFilename(String)} instead
+   * @param n
+   */
+  @Deprecated
+  public void setFileName( String n ) {
+    filename = n;
+  }
+
+  public String getFilename() {
+    return filename;
+  }
+
+  public void setFilename( String n ) {
+    filename = n;
+  }
+
+  public String getRealFilename() {
+    return environmentSubstitute( getFilename() );
+  }
+
+  public String getWorkDirectory() {
+    return workDirectory;
+  }
+
+  public void setWorkDirectory( String n ) {
+    workDirectory = n;
+  }
+
+  public String getScript() {
+    return script;
+  }
+
+  public void setScript( String scriptin ) {
+    script = scriptin;
+  }
+
+  public String getLogFilename() {
+    String retval = "";
+    if ( setLogfile ) {
+      retval += logfile == null ? "" : logfile;
+      Calendar cal = Calendar.getInstance();
+      if ( addDate ) {
+        SimpleDateFormat sdf = new SimpleDateFormat( "yyyyMMdd" );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The SimpleDateFormat object was created without setting the time zone. Make sure that you want to use the default time zone or use letter z, Z or X in the pattern. Otherwise, if you do not call setTimeZone() on the created SimpleDateFormat object, you might get an unexpected date and time when using the object.